ship_tracking/ship_tracking.ipynb

```diff
@@ -0,0 +1,299 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Given a set of colors, let's construct a color key for Datashader to use later, along with a visible legend we can add to such a plot:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "colors    = cc.glasbey_bw_minc_20_minl_30\n",
+    "color_key = {list(groups.keys())[i]:tuple(int(e*255.) for e in v) for i,v in \n",
+    "              enumerate(colors[:(len(groups))][::-1])}\n",
+    "legend    = hv.NdOverlay({groups[k]: hv.Points([0,0], label=str(groups[k])).opts(\n",
+    "                                         color=cc.rgb_to_hex(*v), size=0) \n",
+    "                          for k, v in color_key.items()})\n",
+    "#legend #uncomment to see legend alone"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Load AIS pings\n",
+    "\n",
+    "Next we will load the data from disk, either directly from a spatially indexed Parquet file (if previously cached) or from the raw CSV files. We'll also project the data to the coordinate system we will use later for plotting.\n",
+    "\n",
+    "Since particularly in raw form this is a lot of data, we will use the `map_partitions` functionality of a dask.DataFrame. To do this we define a function to the conversion and an example DataFrame with the required structure:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convert_partition(df):\n",
+    "    east, north = ll2en(df.LON.astype('float32'), df.LAT.astype('float32'))\n",
+    "    return sp.GeoDataFrame({\n",
+    "        'geometry': sp.geometry.PointArray((east, north)),\n",
+    "        'MMSI': df.MMSI.fillna(0).astype('int32'),\n",
+    "        'category': df.VesselType.replace(categories).astype('int32')})\n",
+    "\n",
+    "example = sp.GeoDataFrame({\n",
+    "    'geometry': sp.geometry.PointArray([], dtype='int32'),\n",
+    "    'MMSI':     np.array([], dtype='int32'),\n",
+    "    'category': np.array([], dtype='int32')})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we will define the function that will load our data, reading a much-smaller (and much faster to load) cached Parquet-format file from disk if available:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "basedir = './2020/'\n",
+    "basename = 'AIS_2020_01'\n",
+    "\n",
+    "def load_data():\n",
+    "    cache_file = basedir+basename+\"_gdf.parquet\"\n",
+    "    if os.path.exists(cache_file):\n",
+    "        print(\"Reading Parquet file\")\n",
+    "        return sp.io.read_parquet_dask(cache_file).persist()\n",
+    "\n",
+    "    df = dd.read_csv(basedir+'AIS_2020_01*.csv')\n",
```

Comment:
```suggestion
    "    df = dd.read_csv(basedir+basename+'*.csv', usecols=['MMSI', 'LON', 'LAT', 'VesselType'])
\n",
```

Reply:
Getting another speedup with this, now looks like this on my MBP:

<img width="586" alt="Screen Shot 2021-01-19 at 12 37 43 PM" src="https://user-images.githubusercontent.com/1550771/105029843-2a190400-5a53-11eb-8af4-e5e8e1fb6597.png">